src/NHibernate/Logging.cs

```diff
@@ -349,39 +360,174 @@ public bool IsEnabled(InternalLogLevel logLevel)
 		}
 	}
 
-	public class NoLoggingLoggerFactory: INHibernateLoggerFactory
+	/// <summary>
+	/// Base class for <see cref="INHibernateLogger"/> implementations, provides <see cref="ITransitionalInternaLogger"/>
+	/// methods.
+	/// </summary>
+	// Since 5.1
+	[Obsolete("To be used as logger base class for implementing the obsolete interface ITransitionalInternaLogger until it gets dropped.")]
+	public abstract class NHibernateLoggerBase
+	{
+		private readonly INHibernateLogger _this;
+
+		protected NHibernateLoggerBase()
+		{
+			_this = this as INHibernateLogger ??
+				throw new InvalidOperationException("Concrete implementation must be an INHibernateLogger");
```

Comment:
I have decided to not have this base class directly implementing `INHibernateLogger` in order to avoid having to supply its two new methods in the base class as abstract, which would force them to be virtual.